From 42ea9164463e7c2611365b3812f916ea903bdabf Mon Sep 17 00:00:00 2001
From: fanglifei <fanglifei@eswincomputing.com>
Date: Thu, 23 May 2024 13:14:03 +0800
Subject: [PATCH 15/68] WIN2030-14758: fix(vo): add dma-noncoherent for video
 ouput device

Changelogs:
1. Fix display corruption by adding dma-noncoherent dts proporty

Change-Id: I561ff0990fc293cdf3a2ef765d4d3c0464941c7a
---
 arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi | 1 +
 arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi | 1 +
 2 files changed, 2 insertions(+)

--- a/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi
+++ b/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi
@@ -1739,6 +1739,7 @@
 		video_output: display-subsystem {
 			compatible = "eswin,display-subsystem";
 			ports = <&dc_out>;
+			dma-noncoherent;
 		};
 
 		dvb_widgets: dvb-subsystem {
--- a/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi
+++ b/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi
@@ -1647,6 +1647,7 @@
         d1_video_output: display-subsystem {
             compatible = "eswin,display-subsystem";
             ports = <&d1_dc_out>;
+            dma-noncoherent;
         };
 
         d1_dc: display_control@702c0000 {
